### PR TITLE
[yang-models] Add missing constraints in VLAN yang model

### PR DESCRIPTION
**Pull request by matiAlfaro**
Implement constraints as described in 12256 (sonic-buildimage)

1. **Don't allow write of default vlan (vid = 1)**
- in name leaf - must "not(current() = 'Vlan1')"
- in vlanid leaf - must "not(current() = 1)"
- test VLAN_CREATE_DEFAULT_VLAN

2. **If vlan id already exists, do not allow duplicitous write to ConfigDB**
No changes are required, name is key so it will be unique

3. **Do not allow removal of vlan interface if vlan has IP address assigned**
No chnage is needed since both VLAN_INTERFACE_LIST and VLAN_INTERFACE_IPPREFIX_LIST reference /vlan:sonic-vlan/vlan:VLAN/vlan:VLAN_LIST/vlan:name
- 	Added IP_PREFIX_WITHOUT_CREATNG_VLAN test

4. **Don't allow removal of vlan if there are members assigned**
No chnage is needed since VLAN_MEMBER_LIST refereces "/vlan:sonic-vlan/vlan:VLAN/vlan:VLAN_LIST/vlan:name"
- Added test VLAN_MEMBERS_WITHOUT_CREATING_VLAN

5. **If interface naming mode is alias, make sure corresponding port name exists**
sonic-yang doesn't support adding a ports using alias

6. **Make sure port is not configured as a mirror destination port**
- must "not(/sms:sonic-mirror-session/sms:MIRROR_SESSION/sms:MIRROR_SESSION_LIST[sms:dst_port=current()])" 
- Fixed MIRROR_SESSION in sonic-yang-models\tests\files\sample_config_db.json (Ethernet2 is in Vlan111)
- Test MIRROR_SESSION_ON_VLAN_MEMBER_PORT

7. **Make sure port is not already a vlan member**
No change is needed, <vlan,port> is the key for VLAN_MEMBER_LIST

8. **Make sure port is not a router interface**
- must "not(/intf:sonic-interface/intf:INTERFACE/intf:INTERFACE_LIST[intf:name=current()])"
- Test VLAN_ADD_PORT_THAT_IS_ROUTER_INTERFACE

9. **Make sure port is not already a portchannel member**
- must "not(/lag:sonic-portchannel/lag:PORTCHANNEL_MEMBER/lag:PORTCHANNEL_MEMBER_LIST[lag:port=current()]/lag:name)"
- Test VLAN_ADD_PORT_CHANNEL_MEMBER

10. **Make sure port is not already an untagged member**
- must "not(/vlan:sonic-vlan/vlan:VLAN_MEMBER/vlan:VLAN_MEMBER_LIST[vlan:port=current() and vlan:tagging_mode='untagged' and vlan:name!=current()/../name])"
- Test VLAN_ADD_PORT_THAT_IS_UNTAGGED

**Verify Vlan name matches vlanid**
- must "substring-after(../name, 'Vlan') = current()"
- Test VLAN_CREATE_VLAN_WITH_MISSMATCHING_NAME
##### Work item tracking
- Microsoft ADO **(number only)**:

#### How to verify it
Added tests to verify all modified and added constraints.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/sonic-yang-models/tests/files/sample_config_db.json
Comment (amazor) [inline src/sonic-yang-models/tests/files/sample_config_db.json:1]: Why the change in Ethernet #s in this file?

Comment (matiAlfaro) [reply src/sonic-yang-models/tests/files/sample_config_db.json:1]: 1 & 2 are LAG members
2 is also dst_port in mirror session

File: src/sonic-yang-models/tests/yang_model_tests/tests/vlan.json [hunk 1]
```diff
@@ -96,8 +96,7 @@
         "eStr": "Vlan1 is default Vlan"
     },
     "VLAN_CREATE_VLAN_WITH_MISSMATCHING_NAME": {
-        "desc": "Create Vlan so that name doesn't match id",
-        "eStr": "The vlanid must correspond to the VLAN name"
+        "desc": "Create Vlan so that name doesn't match id"
```
Comment (matiAlfaro) [inline src/sonic-yang-models/tests/yang_model_tests/tests/vlan.json:99]: This is still in-progress
